src/audio/component.c

```diff
@@ -29,31 +29,68 @@ DECLARE_SOF_UUID("component", comp_uuid, 0x7c42ce8b, 0x0108, 0x43d0,
 
 DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 
-static const struct comp_driver *get_drv(uint32_t type)
+static const struct comp_driver *get_drv(struct sof_ipc_comp *comp)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
-	uint32_t flags;
+	struct sof_ipc_comp_new_ext *comp_ext;
 
-	irq_local_disable(flags);
+	if (comp->ext_data_offset &&
+	    comp->hdr.size <
+	    comp->ext_data_offset + sizeof(struct sof_ipc_comp_new_ext)) {
+		tr_err(&comp_tr, "get_drv(): header mismatched! ext_data_offset:0x%x, hdr.size: 0x%x",
+		       comp->ext_data_offset, comp->hdr.size);
+		return NULL;
+	}
+
+	/* use UUID to match the driver if UUID is provided */
+	if (comp->ext_data_offset) {
+		comp_ext = (struct sof_ipc_comp_new_ext *)
+			   ((uint8_t *)comp + comp->ext_data_offset);
+
+		/* use component type if old tplg without UUID used */
+		if (sof_is_uuid_nil(comp_ext->uuid)) {
+			tr_warn(&comp_tr, "UUID is Nil, please check your tplg file, component type will be used\n");
+			goto comp_type_match;
+		}
+
+		/* search driver list with UUID */
+		list_for_item(clist, &drivers->list) {
+			info = container_of(clist, struct comp_driver_info,
+					    list);
+			if (!memcmp(info->drv->uid, comp_ext->uuid,
+				    UUID_SIZE)) {
+				tr_dbg(&comp_tr,
+				       "get_drv_from_uuid(), found driver type %d, uuid %s",
+				       info->drv->type,
+				       info->drv->tctx->uuid_p);
+				drv = info->drv;
+				return drv;
+			}
+		}
+
+		tr_err(&comp_tr, "get_drv(): the provided UUID (%8x%8x%8x%8x) doesn't match to any driver!",
```

Comment:
"the provided UUID (%8x%8x%8x%8x) doesn't match to any driver" better to say "component UUID (%8x%8x%8x%8x) not found"

Reply:
Okay.